src/nvim/drawline.c

```diff
@@ -388,31 +388,33 @@ static void draw_foldcolumn(win_T *wp, winlinevars_T *wlv)
   int fdc = compute_foldcolumn(wp, 0);
   if (fdc > 0) {
     int attr = win_hl_attr(wp, use_cursor_line_highlight(wp, wlv->lnum) ? HLF_CLF : HLF_FC);
-    fill_foldcolumn(wlv, wp, wlv->foldinfo, wlv->lnum, attr, fdc, NULL);
+    fill_foldcolumn(wp, wlv->foldinfo, wlv->lnum, attr, fdc, NULL);
+    assert(wlv->off == 0);
+    wlv->off = fdc;
```

Comment:
This is a weird way to write it, why not just `wlv->off += fdc` which is the actual meaning of the code?

Reply:
`fill_foldcolumn()` uses loop index `i` (which starts at 0) directly.

Reply:
I can read the code myself perfectly fine. The point is that this change is a direct regression in code quality. What was the need to merge it in a panic in the middle of the night?

Reply:
This assert and reliance on absolute position in linebuf was necessary before extending the `fill_foldcolumn()` loop to draw filler `' '`'s instead of calling `draw_coll_fill()`. Could have gotten rid of it after refactoring that, I can make another PR.

Reply:
The entire point of #26528 is that you can now just do changes like:

```diff
diff --git a/src/nvim/drawline.c b/src/nvim/drawline.c
index 0e2f0e95ea..06af183243 100644
--- a/src/nvim/drawline.c
+++ b/src/nvim/drawline.c
@@ -1516,14 +1516,13 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool number_onl
         }
       } else {
         // draw builtin info columns: fold, sign, number
-        draw_foldcolumn(wp, &wlv);
-
         // wp->w_scwidth is zero if signcol=number is used
         for (int sign_idx = 0; sign_idx < wp->w_scwidth; sign_idx++) {
           draw_sign(false, wp, &wlv, sign_idx, sign_cul_attr);
         }
 
         draw_lnum_col(wp, &wlv, sign_num_attr, sign_cul_attr);
+        draw_foldcolumn(wp, &wlv);
       }
 
       win_col_offset = wlv.off;
```

And things would work, because the functions do what they say on the tin. This is nowhere universally true yet but that is the direction we should strive for. Adding **back** random assumptions to helper functions is something I consider a regression.